### PR TITLE
Add Metabase user read access to result_upload_error table

### PR DESCRIPTION
**Pull request by nathancrtr**
## Related Issue
https://app.zenhub.com/workspaces/simplereport-better-data-team-605b43d51ea9f700119a48ee/issues/gh/cdcgov/prime-simplereport/6153

## Changes Proposed
- Grant read privileges to `result_upload_error` table for Metabase user
    - We will mostly/entirely be using Metabase to read from the new `result_upload_error` table

## Additional Information
- None

## Testing
LGTM locally, but I will deploy to a dev environment to confirm.

### CODE REVIEW

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -4825,3 +4825,14 @@ databaseChangeLog:
           - sql:
               sql: |
                 DROP TYPE ${database.defaultSchemaName}.TEST_RESULT_UPLOAD_ERROR;
+  - changeSet:
+      id: result_upload_error-add-no-phi-user
+      author: nathan@skylight.digital
+      comment: Adds permission for the no-PHI user to select from the bulk test result upload errors
+      changes:
+        - tagDatabase:
+            tag:results_upload_error-add-no-phi-user
```
Comment (zdeveloper) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:4834]: can you add a space here

**Review by zdeveloper — APPROVED** [review]

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
Comment (nathancrtr) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml]: Keen eye! 🕵️ added in [212c8d1](https://github.com/CDCgov/prime-simplereport/pull/6155/commits/212c8d1aa174f4f387e97137e116cd84600ec900).

**Comment by zdeveloper** [comment]
waiting on dev3 to be available so we can test it with metabase

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -4825,3 +4825,14 @@ databaseChangeLog:
...
+      id: result_upload_error-add-no-phi-user
+      author: nathan@skylight.digital
+      comment: Adds permission for the no-PHI user to select from the bulk test result upload errors
+      changes:
+        - tagDatabase:
+            tag: results_upload_error-add-no-phi-user
+        - sql:
+            sql: GRANT SELECT ON ${database.defaultSchemaName}.result_upload_error TO ${noPhiUsername};
+      rollback:
+            sql: REVOKE SELECT ON ${database.defaultSchemaName}.result_upload_error FROM ${noPhiUsername};
```
Comment (zdeveloper) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: The rollback block is invalid

**Review by zdeveloper — CHANGES_REQUESTED** [review]

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (nathancrtr) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: Are you looking at an out-of-date commit, maybe? I seem to be able to roll this change back:

```
➜  backend git:(nac/6153-add-result-upload-errors-metabase) ✗ ./gradlew liquibaseRollbackCount -PliquibaseCommandValue=1

> Task :liquibaseRollbackCount
liquibase-plugin: Running the 'main' activity...

Starting Liquibase at 11:39:59 (version 4.9.1 #1978 built at 2022-03-28 19:39+0000)
[2023-07-24 11:40:00] INFO [liquibase.lockservice] Successfully acquired change log lock
[2023-07-24 11:40:00] INFO [liquibase.changelog] Reading from simple_report.databasechangelog
Rolling Back Changeset: db/changelog/db.changelog-master.yaml::result_upload_error-add-no-phi-user::nathan@skylight.digital
[2023-07-24 11:40:01] INFO [liquibase.lockservice] Successfully released change log lock
Liquibase: Rollback has been successful.
```

Comment (zdeveloper) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: hmm... can you verify the access has been revoked?

Comment (zdeveloper) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: This is the error I am seeing
image
and This is what is should be 
image

Comment (nathancrtr) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: I can deploy it out to a Metabase environment to be certain, but looking at the DDL definitions for the table it does seem like the permissions are being changed as expected:

_Migration up_
Screen Shot 2023-07-24 at 11 54 18 AM

_Migration down_
Screen Shot 2023-07-24 at 11 53 26 AM

@zdeveloper what are you seeing?

Comment (zdeveloper) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: hmm I am a bit puzzled, I think its removing the granting migration, thats why you dont see `grant select on result_upload_error to simplereport_no_phi` in the migration down DDL

I would expect it to have the revoke statement in the DDL in the migration down


`REVOKE SELECT ON result_upload_error FROM simple_report_no_phi`

Comment (zdeveloper) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:4838]: And also my understanding how the DDL for permissions was wrong, I expected it to have the grant and the revoke statement in the DDL, but I guess liquibase was able to eliminate them both from the DDL.

**Review by DanielSass — APPROVED** [review]

**Pull request merged**